test/StringField_test.js

```diff
@@ -209,17 +198,19 @@ describe("StringField", () => {
     });
 
     it("should reflect the change into the dom", () => {
-      const {node} = createFormComponent({schema: {
+      const {comp, node} = createFormComponent({schema: {
         type: "string",
         format: "date-time",
       }});
 
-      const newDatetime = new Date().toJSON();
-      Simulate.change(node.querySelector("[type=datetime-local]"), {
-        target: {value: newDatetime}
-      });
+      Simulate.change(node.querySelector("#root_year"), {target: {value: "2010"}});
+      Simulate.change(node.querySelector("#root_month"), {target: {value: "12"}});
+      Simulate.change(node.querySelector("#root_day"), {target: {value: "1"}});
+      Simulate.change(node.querySelector("#root_hour"), {target: {value: "0"}});
+      Simulate.change(node.querySelector("#root_minute"), {target: {value: "0"}});
+      Simulate.change(node.querySelector("#root_second"), {target: {value: "0"}});
 
-      expect(node.querySelector("[type=datetime-local]").value).eql(newDatetime);
+      expect(comp.state.formData).eql("2010-12-01T00:00:00.000Z");
```

Comment:
Maybe also add a test to make sure that all the options are present in the selects? It would be unusable if for some reason we render selects which are empty, or missing some values...